app/src/main/java/com/github/jorgecastillo/kotlinandroid/data/HeroesRepository.kt

```diff
@@ -2,7 +2,11 @@ package com.github.jorgecastillo.kotlinandroid.data
 
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchAllHeroes
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchHeroesFromAvengerComics
+import com.karumi.marvelapiclient.model.CharacterDto
+import kategory.HK
 
-fun getHeroesWithCachePolicy() = fetchAllHeroes()
+inline fun <reified F> getHeroesWithCachePolicy(): HK<F, List<CharacterDto>> =
```

Comment:
As I see it, the need to use `reify` here is imposed by our internal behind the scenes implementation of how to provide defaults for global instances. The user is changing his way to code functions imposed not by function requirements but by how our lib is written behind the scenes around some internals. For this case, using an inline reified function by a user of the lib is not really giving any clue of what you are going to achieve with it, and to me it feels missleading.

Before this PR, I was already using "implicits" (global instances) on the Kleisli without using inline or reified. The `Id()` was being injected and it was working as expected. You can check `nested-monads` branch for it. So my question is, what's the real enhancement with this new way to use global instances?